modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/builder/SchemaBuilder.java

```diff
@@ -889,23 +907,38 @@ private boolean shouldGeneratePojoChildElements(Class<?> type)
             private boolean shouldForceOptional(Class<?> type)
             {
                 return !parameterModel.isRequired() ||
-                       !subTypesMapping.getSubTypes(parameterModel.getType()).isEmpty()||
+                       !subTypesMapping.getSubTypes(parameterModel.getType()).isEmpty() ||
                        (IntrospectionUtils.isInstantiable(type) && ExpressionSupport.REQUIRED != parameterModel.getExpressionSupport());
             }
         };
     }
 
-    private void registerPojoSubtypes(List<MetadataType> subTypes , ExplicitGroup all)
+    private void addImportedTypeRef(Class<?> extensionType, ParameterModel parameterModel, ExplicitGroup all)
+    {
+        XmlModelProperty xml = new XmlModelEnricher().createXmlModelProperty(
```

Comment:
if you need to reuse logic extract to a utils... do not put a ModelEnricher in here